### ISSUE TITLE
Missing docs or invalid data related to debug information

### ISSUE THREAD

**Issue by rolandshacks** [cc65/cc65#1969]
I am currently working on deep integration of CC65 to visual studio code, including a debugger extension supporting both asm and C. While code level debugging works already quite well, I do not see clear information about how to handle variables on the stack. With reverse engineering, I found out that the stack-pointer is $02-$03 and csyms allow (not convenient, but it seems to work) lookup of variables on the stack. BUT: there is no type information, the only information is offset compared to the parent span.

Example snippet:

uint8_t test_function(uint8_t arg1, uint16_t arg2, uint8_t arg3, uint8_t arg4) {
	uint8_t result = arg1 + arg2 + arg3 + arg4;
	return result;
}

int main() {
	int a = test_function(global_value1, (uint8_t) global_value2, 3, 4);
    return 0;
}

^^^How to display the value of arg1...arg4 in case the program reached a breakpoint within test_function(...) ?

Thanks!

**Comment by kugelfuhr** [comment]
Short answer: You cannot.

Slightly longer answer: The debug info is incomplete. It contains the name and stack offset of local symbols and that's it. The type is always "void" and there is no correction for the offset in each span. You can test that using your code and  the dbgsh program:

```
$ dbgsh load test.dbg
File loaded successfully
dbgsh> show csymbol 1 2 3 4
  id  name                        type  kind   sc   offs  symbol scope
---------------------------------------------------------------------------
   1  arg1                           0     1     0     4       -     1
   2  arg2                           0     1     0     2       -     1
   3  arg3                           0     1     0     1       -     1
   4  arg4                           0     1     0     0       -     1
dbgsh> show type 0
  id  description
---------------------------------------------------------------------------
   0  VOID
dbgsh> 
```

**Comment by rolandshacks** [comment]
Thanks for the quick answer. ... that's really unfortunate. I am so close
the provide great debugging experience within vscode... (I mean, it is
already great to navigate, step, etc., ...).

Are there any plans to complete and improve the debug info? Also, anything
like an AST giving better and easier structural information would be great.

Regards,
Roland


Am Fr., 13. Jan. 2023 um 11:18 Uhr schrieb kugelfuhr <
***@***.***>:

**Comment by kugelfuhr** [comment]
I'm the wrong one to answer this question. You may look at [the change history of the debug info module](https://github.com/cc65/cc65/commits/master/src/dbginfo) and then estimate your chances.

**Comment by mrdudz** [comment]
Yeah if that is what you need/want, your best bet is to get cracking and start fixing it :) There is noone working on this right now.
